### ISSUE TITLE
TX16S, EdgeTX 2.10.x, no RC Link using "External RF" JR Bay (external module) and PPM\SBUS

### ISSUE THREAD

**Issue by packetkiller** [EdgeTX/edgetx#5514]
### Is there an existing issue for this problem?

- [X] I have searched the existing issues

### What part of EdgeTX is the focus of this bug?

Transmitter firmware

### Current Behavior

Using TX16S with RFD TXMOD v2 in JR port to a RFD 900x v2 to a Matek H743-WING. TX16S running 2.10.2 or 2.10.4. Flight Controller running Ardupilot 4.5.4 (Plane). Radio receives telemetry from FC and telemetry is viewable in Mission Planner via Mavlink from TXMOD WiFi. Telemetry is NOT visible in TX16S. RC link is not transmitted from TX16S to TXMOD (and thus never shows up on air side). Validated radios on both sides TX/RX sending and recieving data using scope. RFD radios are using default settings except for NETID custom value that is set to same on both.

### Expected Behavior

RC link sent to air side and then to FC. Telemetry from FC sent to TXMOD, then to TX16S, then discoverable in EdgeTX and live stream via telemetry LUA widget.

### Steps To Reproduce

Upgrade working firmware and SD card contents of EdgeTX from 2.9.x to 2.10.x.

### Version

2.10.4

### Transmitter

RadioMaster TX16S / TX16SMK2

### Anything else?

This issue seems to be a root cause for issue #5064 (telemetry only issue).

**Comment by pfeerick** [comment]
> RC link is not transmitted from TX16S to TXMOD 

This sounds somewhat fishy... shouldn't it be configured simply as PPM output? So if you have PPM output on the top pin of the module bay... you should at least have the control link, thus meaning that whatever the core issue is here, it appears that most of the reports in 5064 are common to the RFD TXMOD (v2 or modified v1), and changes around PPM/telemetry. Which would probably be that it defaults to No telemetry (as PPM doesn't have telemetry), and an option to allow MLink compatible telemetry... we probably need another option to add SPort telemetry...

**Comment by packetkiller** [comment]
Even with the TX16S EdgeTX "External RF" option configured with PPM, MLink Telemetry, Channel Range CH1 to CH16, PPM frame set to 38.5ms, 300us, and -, there is no RC link but there is telemetry coming back to the TXMOD and viewable via the TXMOD WiFi in Mission Planner, but not via the TX16S EdgeTX (no telemetry discoverable). Powerup sequence is model, then TX16S, then connect to TXMOD via laptop WiFi.

**Comment by mha1** [comment]
TX16s External RF set to PPM/No Telemetry or PPM/MLink, CH1..CH16, 38.5ms, 300us. -. A perfect 16ch PPM stream at the JR Bay CPPM pin. You've gotta find your problem at the receiving end of the PPM stream.

![image](https://github.com/user-attachments/assets/7cf8e3a7-3d7a-4ac3-a40b-862f11285f1e)

**Comment by packetkiller** [comment]
> TX16s External RF set to PPM/No Telemetry or PPM/MLink, CH1..CH16, 38.5ms, 300us. -. A perfect 16ch PPM stream at the JR Bay CPPM pin. You've gotta find your problem at the receiving end of the PPM stream.
> 
> ![image](https://private-user-images.githubusercontent.com/5615068/365393547-7cf8e3a7-3d7a-4ac3-a40b-862f11285f1e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU3NDA5NTIsIm5iZiI6MTcyNTc0MDY1MiwicGF0aCI6Ii81NjE1MDY4LzM2NTM5MzU0Ny03Y2Y4ZTNhNy0zZDdhLTRhYzMtYTQwYi04NjJmMTEyODVmMWUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkwNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MDdUMjAyNDEyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NTE2MjFjNzc2Njc3ZWY1ODM4MWYzMTU2ZWQ1M2ZmYjM2OWYyNDM2NmNhZWNiYWQzNDRmY2ZlNWMyMWRmNjgzYyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.S0-W8OMeqSz_e4a1HBNdwLfOqtiJm3LE87MuLjh8OUQ) ![image](https://private-user-images.githubusercontent.com/5615068/365394173-724b0e3e-4d63-435d-97da-4ad03f99e3bc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU3NDA5NTIsIm5iZiI6MTcyNTc0MDY1MiwicGF0aCI6Ii81NjE1MDY4LzM2NTM5NDE3My03MjRiMGUzZS00ZDYzLTQzNWQtOTdkYS00YWQwM2Y5OWUzYmMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkwNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MDdUMjAyNDEyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MmE4NjQxYmViNGJjYjQyYmNiZDM2ZmFjOTJkZWNlMzc0ZDY0NGRiZTg1OWNjZDNjZTdlM2U2MjQyZGQzNjJlOCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.aGBD0Qr4wcG7EW6m7ZNsrMHDXJ0gDUx0WKP7TmVFz9o)

I agree. I will check each leg of the radios starting at the TX16S and working my way to the FC. The first test at the TX16S JR bay on PPM pin shows good PPM signal coming out. When I move the sticks, I see the signal move. Will status each leg to show others how I troubleshoot this...maybe it will help someone else. Next checking the 900x side to see if it got the PPM signal.

**Comment by packetkiller** [comment]
So first leg across the radios shows no signal on pin 15 as expected. So the radios are talking as I can configure both from the TXMOD. For this forum, this means that the TX16S is working and spitting out PPM to the radio but the radio is not relaying that to the other side or the other side is not picking it up. I will keep working this with 3DR and report back when I have something to report.

**Comment by packetkiller** [comment]
Update, managed to get ppm across link by resetting both radios to default, then setting S16:GPI1_1R/CIN = 1 on the TXMOD radio. Though I had this setting in place before and it did not work, this time, I did not run the "Go to First Run Wizard!". I have confirmed that the wizard sets something that breaks a standard PPM passthrough link. I am now going to attempt to validate on a flight controller and not just the oscilloscope.

**Comment by packetkiller** [comment]
Update, am not able to get the flight controller to recognize the PPM signal. Going to try and move from PPM to SBUS and try again. RFDesign support has validated the PPM configuration. I have asked for config changes required to enable SBUS. Will see if this helps.

**Comment by packetkiller** [comment]
Update, have now narrowed the issue to the radio software. Once a pair of radios are linked and configured, using the TXMOD web interface to change the configuration seems to "get stuck". I have been able to use the USB interface to change the radio settings but even that can get stuck and take a few tries to make a single change. Once changed, the pair do not communicate, thus the issue. Will continue to work with RFD to sort it out.

**Comment by pfeerick** [comment]
Thank you for the updates, good to know making progressnarrowing the culprit, whatever it may be

**Comment by packetkiller** [comment]
Finally figured it out. The main issue was related to the nature of how the radios save data to memory...meaning you must follow strict procedure for certain configuration changes. In short, when in doubt, just start back from all defaults and reconfigure fresh. It is a bit late here so will add another comment with the configuration details in the morning.

**Comment by mha1** [comment]
Good to know. I was certain you'd find the problem at the receiving end of the PPM stream. Don't forget to close this issue.
